### ISSUE TITLE
Code for PPA with different losses?

### ISSUE THREAD

**Issue by ShailenSmith** [LukasStruppek/Plug-and-Play-Attacks#17]
Hi,

We are hoping to experiment with PPA replacing the Poincare loss with both standard CE and regularized logit identity loss, as done in App. C.7 in the "Be Careful What You Smooth For" paper. Are you able to share the code you used to do this?

If not, were there any tricky/specific design choices you had to make during implementation? Otherwise we should be good to implement it ourselves.

Thanks for your help!

**Comment by LukasStruppek** [comment]
Hey,

using CE instead of Poincaré is quite simple: Just replace the Poincaré loss with the standard cross-entropy loss function of PyTorch in this line:
https://github.com/LukasStruppek/Plug-and-Play-Attacks/blob/ca6e390565200fe05aeb228f8e342365de70004d/attacks/optimize.py#L44C1-L45C1

For the regularized logits, we used the official implementation:
https://github.com/sutd-visual-computing-group/Re-thinking_MI

Does this solve your issue?
